Core/Inc/monitor.h

```diff
@@ -29,29 +38,21 @@ void vIMUMonitor(void *pv_params);
 extern osThreadId_t imu_monitor_handle;
 extern const osThreadAttr_t imu_monitor_attributes;
 
-/* Task for Monitoring the TSMS sense on PDU  CTRL expander*/
-void vTsmsMonitor(void *pv_params);
-extern osThreadId_t tsms_monitor_handle;
-extern const osThreadAttr_t tsms_monitor_attributes;
-
-/* Task for Monitoring the Fuses on PDU */
-void vFusingMonitor(void *pv_params);
-extern osThreadId_t fusing_monitor_handle;
-extern const osThreadAttr_t fusing_monitor_attributes;
-
 /* Task for Monitoring the Shutdown Loop */
 void vShutdownMonitor(void *pv_params);
 extern osThreadId_t shutdown_monitor_handle;
 extern const osThreadAttr_t shutdown_monitor_attributes;
 
-void vSteeringIOButtonsMonitor(void *pv_params);
-extern osThreadId_t steeringio_buttons_monitor_handle;
-extern const osThreadAttr_t steeringio_buttons_monitor_attributes;
-
-void vBrakelightMonitor(void *pv_params);
-extern osThreadId brakelight_monitor_handle;
-extern const osThreadAttr_t brakelight_monitor_attributes;
-
-bool get_tsms();
+/* Arguments for the data collection thread */
+typedef struct {
+	mpu_t *mpu;
+	pdu_t *pdu;
+	steeringio_t *wheel;
+} data_collection_args_t;
```

Comment:
im not a huge fan of putting these in a struct - I think thats overkill. Any particular reason you prefer them this way?

Reply:
You don't have to deal with the order of elements in an array and freeing the struct pointer is easier.